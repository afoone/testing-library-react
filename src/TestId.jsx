import React from 'react';

const TestId = () => {
  return <div>
    <p>
        <span data-testid="search-label">Search:</span>
        <input data-testid="search-input" type="text" placeholder="Search" />
        <span data-testid="search-results" className='pepe'>Searches for...</span>
    </p>
  </div>;
};

export default TestId;
