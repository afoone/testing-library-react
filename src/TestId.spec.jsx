import { screen, render } from "@testing-library/react";
import TestId from "./TestId";

const setup = () => {
  render(<TestId />);
};

describe("TestId", () => {
  beforeEach(() => {
    setup();
  });
  it("renders", () => {
    const { container } = render(<TestId />);
    //expect(container).toMatchSnapshot();
  });

  it("renders search label", () => {
    expect(screen.getByText("Search:")).toBeInTheDocument();
    expect(screen.getByTestId("search-label")).toBeInTheDocument();
  });

  it("renders search input", () => {
    expect(screen.getByPlaceholderText(/search/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/search/i)).toHaveAttribute(
      "data-testid",
      "search-input"
    );
  });

  it("renders search results", () => {
    expect(screen.getByTestId("search-results")).toBeInTheDocument();
    expect(screen.getByText("Searches for...")).toHaveAttribute(
      "class",
      "pepe"
    );
  });

  test("theres no heading", () => {
    expect(screen.queryByRole('heading')).not.toBeInTheDocument();
  });
});
