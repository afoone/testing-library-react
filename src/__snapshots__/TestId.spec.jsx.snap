// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestId renders 1`] = `
<div>
  <div>
    <p>
      <span
        data-testid="search-label"
      >
        Search:
      </span>
      <input
        data-testid="search-input"
        placeholder="Search..."
        type="text"
      />
      <span
        data-testid="search-results"
      >
        Searches for...
      </span>
    </p>
  </div>
</div>
`;
